de.fu_berlin.inf.dpp.ui.frontend/html/src/components/ToolBar/ConnectionSplitButton.jsx

```diff
@@ -0,0 +1,51 @@
+import { Account } from 'Utils/propTypes'
+import { MenuItem, SplitButton } from 'react-bootstrap'
+import { PropTypes as PM } from 'mobx-react'
+import { Text } from 'react-localize'
+import { getJid, noop } from 'Utils'
+
+import React from 'react'
+
+const ConnectionSplitButtonProps = {
+  accounts: PM.observableArrayOf(Account).isRequired
+}
+
+class ConnectionSplitButton extends React.Component {
+  render () {
+    const connectionStateMap = {
+      'INITIALIZING': { onClick: this.props.onConnect, messageId: 'action.connect' },
+      'NOT_CONNECTED': { onClick: this.props.onConnect, messageId: 'action.connect' },
+      'CONNECTED': { onClick: this.props.onDisconnect, messageId: 'action.disconnect' },
+      'ERROR': { onClick: noop, messageId: 'action.connectionError' },
+      'CONNECTING': { onClick: noop, messageId: 'action.connecting' },
+      'DISCONNECTING': { onClick: noop, messageId: 'action.disconnecting' }
+    }
```

Comment:
I am not very familiar with how react updates it's contents internally, going from my naive assumption that react would basically call the render() function, it seems that `connectionStateMap` would be re-created every time even though nothing in its contents changes, maybe move it to the constructor?

The thing that I am worried about is that the internal browser in Eclipse on some machines is really old (IE9 on my machine at least), and since babel would transpile the code, it would even less efficient.